src/networkextension.cs

```diff
@@ -1109,7 +1111,7 @@ interface NETunnelProvider
 	{
 		[Export ("handleAppMessage:completionHandler:")]
 		[Async]
-		void HandleAppMessage (NSData messageData, [NullAllowed] Action<NSData> completionHandler);
+		void HandleAppMessage (NSData messageData, [NullAllowed] Action<NSData?> completionHandler);
```

Comment:
Congrats on your first pr!
I believe NullAllowed is not necessary here, you can use the ? suffix instead. Like so: 
```suggestion
		void HandleAppMessage (NSData messageData, Action<NSData?>? completionHandler);
```

Reply:
@haritha-mohan we need the attr else the generator will not output the correct data.

Reply:
This is what I get when I remove [NullAllowed] and change it to Action<NSData?>?:

public unsafe virtual void HandleAppMessage (NSData messageData, [BlockProxy (typeof (ObjCRuntime.Trampolines.NIDActionArity1V51))]global::**System.Action<NSData>** completionHandler)

Reply:
> If the reference type does not have this attribute, the binding tool will generate a check for the value being assigned before passing it to Objective-C and will generate a check that will throw an ArgumentNullException if the value assigned is null.
https://learn.microsoft.com/en-us/xamarin/cross-platform/macios/binding/binding-types-reference#nullallowedattribute

Does the generated null check differentiate between the two situations?

Reply:
@tj-devel709 the generated code is wrong, it is ignoring the ? annotation of the Action generic.